src/scripts/book-search.ts

```
@@ -0,0 +1,110 @@
+  // Type assertions after null check - TypeScript now knows these are not null
+  const searchInputElement = searchInput as HTMLInputElement;
+  const genreFilterElement = genreFilter as HTMLSelectElement;
+  const tagFilterElement = tagFilter as HTMLSelectElement;
+  const clearFiltersBtnElement = clearFiltersBtn as HTMLButtonElement;
+  const resetSearchBtnElement = resetSearchBtn as HTMLButtonElement;
+  const booksGridElement = booksGrid as HTMLElement;
+  const emptyStateElement = emptyState as HTMLElement;
+  const resultsCountElement = resultsCount as HTMLElement;
+
+  // Debounce function for search input
+  function debounce(func: Function, wait: number) {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)